python/pyarrow/_parquet_encryption.pyx

```diff
@@ -300,20 +306,29 @@ cdef class KmsConnectionConfig(_Weakrefable):
 
 # Callback definitions for CPyKmsClientVtable
 cdef void _cb_wrap_key(
-        handler, const c_string& key_bytes,
+        handler, const CSecureString& key,
         const c_string& master_key_identifier, c_string* out) except *:
+    cdef:
+        cpp_string_view view = key.as_view()
+    key_bytes = PyObject_to_object(
+        PyBytes_FromStringAndSizeNative(view.data(), view.size()))
```

Comment:
Of course this means the `bytes` object payload won't be cleared securely like the `SecureString` is.

We could try to tackle this as a followup issue/PR. `bytes` objects being immutable, this might be a bit delicate...